drake/systems/plants/collision/bullet_model.cc

```diff
@@ -637,17 +639,15 @@ bool BulletModel::closestPointsAllToAll(
     const std::vector<ElementId>& ids_to_check, const bool use_margins,
     std::vector<PointPair>& closest_points) {
   std::vector<ElementIdPair> id_pairs;
-  for (auto idA_iter = ids_to_check.begin(); idA_iter != ids_to_check.end();
-       ++idA_iter) {
-    auto elementA_iter = elements.find(*idA_iter);
-    if (elementA_iter != elements.end()) {
-      for (auto idB_iter = idA_iter + 1; idB_iter != ids_to_check.end();
-           ++idB_iter) {
-        auto elementB_iter = elements.find(*idB_iter);
-        if (elementB_iter != elements.end()) {
-          if (elements[*idA_iter]->collidesWith(elements[*idB_iter].get())) {
-            id_pairs.push_back(std::make_pair(*idA_iter, *idB_iter));
-          }
+  for (int i = 0; i < ids_to_check.size(); ++i) {
```

Comment:
I certainly hope not -- there are many occasions where single letter variables create the clearest code (they are often i,j,k and x,y,z). As evidence that they are not forbidden by the Google Style Guide, check out [this example](https://google.github.io/styleguide/cppguide.html#Implementation_Comments):
```c++
// Divide result by two, taking into account that x
// contains the carry from the add.
for (int i = 0; i < result->size(); i++) {
  x = (x << 8) + (*result)[i];
  (*result)[i] = x >> 1;
  x &= 1;
}
```
That is given as an example of good commenting the style guide, but hopefully they wouldn't use nonconformant code there.